### DISCUSSION TITLE
Huawei R4850G2 - instabile CAN Verbindung

### DISCUSSION THREAD

**Discussion by sabafreiburg** [hoylabs/OpenDTU-OnBattery#431] [Q&A]
Ich bin mit einem Release aus Juli gestartet und mir fällt seit Beginn die instabile Kommunikation mit dem Huawei R4850G2 auf. 
Es gibt zeitweise keine Kommunikation für 30....60....150 Sekunden und dann wieder eine ganze Weile alles ok.

Auf den Tip hier, habe ich ältere Release probiert
[https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/299#discussioncomment-6809509](url)

Bis zum release 2023.06.26 funktioniert die Kommunikation zuverlässig, "letzte Aktualisierung" meistens 5sec, selten 10sec.

**Comment by eu-gh** [comment]
Das Thema ist in den Issues schon öfters adressiert worden, siehe z.B:
https://github.com/helgeerbe/OpenDTU-OnBattery/issues/387
https://github.com/helgeerbe/OpenDTU-OnBattery/issues/316
und daraus resultierend:
https://github.com/helgeerbe/OpenDTU-OnBattery/issues/381

Ich habe aus diesem Grund auch sehr lange ein altes Release (glaube [2023.07.24](https://github.com/helgeerbe/OpenDTU-OnBattery/releases/tag/2023.07.24)) verwendet und zusätzlich die date_age von Pylontech und Huawei inventarisiert. Ein exemplarischer Tag damit:
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/141823622/19570cb7-f98a-4cb7-87d5-fa7bafd2a767)

Das Ganze ist mit Release [2023.09.13](https://github.com/helgeerbe/OpenDTU-OnBattery/releases/tag/2023.09.13) aber wesentlich besser geworden. Wenn ich das richtig verstanden habe, wurde die Huawei CAN-Kommunikation nun in einen anderen Thread (oder sogar auf einen anderen Core?) ausgelagert. Zum Vergleich der gestrige Tag:
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/141823622/d1e3d80c-d956-46ef-8720-563dc999d76b)

Es macht aber auch einen Unterschied, wie die sonstigen Parameter stehen. z.B. MQTT- und Inverter-Polling-Intervall. Wenn hier überall relativ kleine Werte eingestellt werden, kommt die ESP-Plattform insgesamt auch mal an ihre Grenzen.
